File: example/src/main/java/com/crowdin/platform/example/fragments/GalleryFragment.java
package com.crowdin.platform.example.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.crowdin.platform.example.R;
import com.crowdin.platform.example.adapter.SampleAdapter;

public class GalleryFragment extends Fragment {

    public static GalleryFragment newInstance() {
        return new GalleryFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_gallery, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        RecyclerView recyclerView = view.findViewById(R.id.recycler_view);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));

        String[] array = getResources().getStringArray(R.array.string_array);
        RecyclerView.Adapter adapter = new SampleAdapter(array);
        recyclerView.setAdapter(adapter);
    }
}

File: example/src/main/java/com/crowdin/platform/example/fragments/GalleryFragment.kt
package com.crowdin.platform.example.fragments

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.crowdin.platform.Crowdin
import com.crowdin.platform.LoadingStateListener
import com.crowdin.platform.example.R
import com.crowdin.platform.example.adapter.SampleAdapter

class GalleryFragment : Fragment(), LoadingStateListener {

    private lateinit var adapter: SampleAdapter
    private lateinit var recyclerView: RecyclerView

    companion object {
        fun newInstance(): GalleryFragment = GalleryFragment()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_gallery, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        recyclerView = view.findViewById(R.id.recycler_view)
        recyclerView.setHasFixedSize(true)
        recyclerView.layoutManager = LinearLayoutManager(context)
        val array = resources.getStringArray(R.array.string_array)
        adapter = SampleAdapter(array)
        recyclerView.adapter = adapter

        view.findViewById<TextView>(R.id.textView3).setOnClickListener { Crowdin.sendScreenshot(activity!!) }
        Crowdin.registerDataLoadingObserver(this)
    }

    override fun onDataChanged() {
        val array = resources.getStringArray(R.array.string_array)
        adapter.updateData(array)
        Log.d("Crowdin", "GalleryFragment: onSuccess")
    }

    override fun onFailure(throwable: Throwable) {
        Log.d("Crowdin", "GalleryFragment: onFailure ${throwable.localizedMessage}")
    }

    override fun onDestroy() {
        super.onDestroy()
        Crowdin.unregisterDataLoadingObserver(this)
    }
}